pagopa/io-app
# feat(Piattaforma Notifiche): [IAMVL-83] Show the timeline in a PN message

## Short description
### ⚠️ **This PR depends on #4047**

This branch implements the timeline in a PN message details.

| Default | Expanded | 
| - | - |
| ![Collapsed](https://user-images.githubusercontent.com/467098/179699887-933f87af-97bb-4ce5-8592-5b201cecf9f2.png) | ![Expanded](https://user-images.githubusercontent.com/467098/179699936-f678a9f6-6cd5-4a31-b501-7b0e1fe4ed65.png) |

**Beware that a CTA below the timeline to open the PN website is still missing and will be added with a following PR, because it requires a modification to `io-services-metadata`**.

## List of changes proposed in this pull request
- Parse the `NotificationStatusHistory` from the PN message content
- Add `PnMessageTimeline` to the `PnMessageDetails`
- ⚠️ I had to port `UTCISODateFromString` from the latest `io-ts-commons` because PN for the timeline uses the timezone defined with +/-00:00 format, and the type defined in `italia-ts-commons` used by the app doesn't support it. This port has been added to `ts/features/pn/utils/dates.ts` and it's only used by the custom PN decoder.

## How to test
Use the [IIP-18-add-pn-endpoints](https://github.com/pagopa/io-dev-api-server/tree/IIP-18-add-pn-endpoints) branch of the dev server and add this to your `config.json`:
```
{
  "messages": {
    "pnCount": 1
  },
  "services": {
    "includePn": true
  }
}
```
Set `PN_ENABLED=YES` to `.env` and run the app, then open a message from PN and verify that the timeline is shown.
